grip_api/tests/code_editors/test_yaml_code_editor.py

```diff
@@ -0,0 +1,251 @@
+    """
+    set_text_and_check_parsing_result(yaml_editor, input_text, wrong_lines, parsed_dict)
+
+
+@pytest.mark.parametrize("input_text, wrong_lines, parsed_dict",
+                         [("# The first line is a comment\nBut the second one is not", [1], {}),
+                          ("# If the whole text is\n# made of comments", [], {}),
+                          ("key: 12 # Adding a comment to a valid line must not change its parsing", [], {"key": 12}),
+                          ("\ninvalid_line # A comment should not make an non valid line valid", [1], {})])
+def test_editor_parsing_comments(yaml_editor, input_text: str, wrong_lines: List[int],
+                                 parsed_dict: Dict[str, Union[int, float, bool, str, List, Dict]]) -> None:
+    """
+        Test that commented lines don't throw the parser off, i.e. don't change its behavior
+
+        @param yaml_editor: Fixture previously defined providing the method with an instance of the object to be tested
+        @param input_text: Text to set to the code editor and that will be parsed
+        @param wrong_lines: List of the indices of the lines wrongly formatted
+        @param parsed_dict: Dictionary only containing the parsed version of the properly formatted components
+    """
+    set_text_and_check_parsing_result(yaml_editor, input_text, wrong_lines, parsed_dict)
+
+
+@pytest.mark.parametrize("input_text, wrong_lines, parsed_dict",
+                         [("key: [1, True, -8.5, value]", [], {"key": [1, True, -8.5, "value"]}),
+                          ("key: [1, True,\n-8.5, value]", [0, 1], {}),
+                          ("key:\n  [1, True, -8.5, value]", [0, 1], {"key": {}}),
+                          ("key:\n  -1\n  -True\n  --8.5\n  -value", [], {"key": [1, True, -8.5, "value"]}),
+                          ("key:\n  - 1\n  - True\n  - -8.5\n  - value", [], {"key": [1, True, -8.5, "value"]})])
+def test_editor_parsing_lists(yaml_editor, input_text: str, wrong_lines: List[int],
+                              parsed_dict: Dict[str, Union[int, float, bool, str, List, Dict]]) -> None:
+    """
+        Test that lists are properly handled when set as values of a dictionary
+
+        @param yaml_editor: Fixture previously defined providing the method with an instance of the object to be tested
+        @param input_text: Text to set to the code editor and that will be parsed
+        @param wrong_lines: List of the indices of the lines wrongly formatted
+        @param parsed_dict: Dictionary only containing the parsed version of the properly formatted components
+    """
+    set_text_and_check_parsing_result(yaml_editor, input_text, wrong_lines, parsed_dict)
+
+
+@pytest.mark.parametrize("input_text, wrong_lines, parsed_dict",
+                         [("key: {subkey1 : value, subkey2 : -1.5, subkey3 : False , sub: [-1, 2.6, True], -8: test}",
```

Comment:
Not sure how you want to add indentations...

Reply:
Me neither, but maybe something like this would work?
```
@pytest.mark.parametrize("input_text, wrong_lines, parsed_dict",
                         [("key:
                            {subkey1 : value,
                             subkey2 : -1.5,
                             subkey3 : False ,
                             sub: [-1, 2.6, True],
                             -8: test}",
                            [],
                            {"key":
                             {"subkey1": "value",
                              "subkey2": -1.5,
                              "subkey3": False,
                              "sub": [-1, 2.6, True],
                              -8: "test"}}),
                            (...remaining_tests...)
```

Reply:
not a huge fan at all! For me that's even more unreadable, but if you insist I can do it